ansible/all-in-one/roles/xsnippet/tasks/main.yml

```diff
@@ -54,7 +34,80 @@
 - file:
     path: "/home/xsnippet/.config/systemd/user"
     state: directory
+- file:
+    path: "/home/xsnippet/.config/letsencrypt"
+    state: directory
+  when: xsnippet_api_https_enabled or xsnippet_spa_https_enabled
 
+- name: prepare the certbot config
+  template:
+    src:  certbot.ini.j2
+    dest: "/home/xsnippet/.config/letsencrypt/cli.ini"
+    mode: 0600
+    owner: xsnippet
+    group: xsnippet
+  when: xsnippet_api_https_enabled or xsnippet_spa_https_enabled
+
+- name: prepare the certbot-dns-digitalocean config
+  template:
+    src:  dns-digitalocean-credentials.ini.j2
+    dest: "/home/xsnippet/.config/letsencrypt/dns-digitalocean-credentials.ini"
+    mode: 0600
+    owner: xsnippet
+    group: xsnippet
+  when: xsnippet_api_https_enabled or xsnippet_spa_https_enabled
+
+- name: generate ssl for {{ xsnippet_api_server_name }}
+  command: certbot certonly --domains {{ xsnippet_api_server_name }}
+  when: xsnippet_api_https_enabled
+
+- name: generate ssl for {{ xsnippet_spa_server_name }}
+  command: certbot certonly --domains {{ xsnippet_spa_server_name }}
+  when: xsnippet_spa_https_enabled
```

Comment:
**n.b.*: Whenever we run the playbook, this task is going to be executed. :( The only way to prevent this is to have the same block we have below (checking the certs) but above. We would need to have it twice because below we are trying to resolve symlinks.

Reply:
Would https://docs.ansible.com/ansible/latest/collections/ansible/builtin/command_module.html#parameter-creates help us here? IIUC, we only need to check if the symlinks exist, not the actual files.

Reply:
Sounds good. Thank you.